### DISCUSSION TITLE
Caching Gtag events on server

### DISCUSSION THREAD

**Discussion by RomanSkrypnik** [johannschopplich/nuxt-gtag#21] [Ideas]
As we know it's not possible to obtain `window` object on server side
So, is that possible to cache all fired events on server and append them to `dataLayer` when client side is rendered?
For example: `useState` - if we use `useState` on server side, its last assigned value will be saved on client side also
If it's not possible, could you please describe why?

**Comment by johannschopplich** [comment]
That's right, Gtag.js itself can't be used on the server. The composables provided by this module are SSR-safe. Meaning that they won't have an effect server-side but will trigger and event in the client. Collecting events and appending them after hydration doesn't make sense to me, because e.g. `useTrackEvent` will only have an effect in the client, not on the server:

```ts
// This will have no effect on the server, but track the event
// in the client after Nuxt hydration
useTrackEvent('login')
```
